File: src/entities/ImdbMovie.ts
import { Column, Entity, Index, OneToMany } from "typeorm";
import { Movie } from "./Movie";

@Index("pk_imdb_movie", ["imdbId"], { unique: true })
@Entity("imdb_movie", { schema: "dionysus" })
export class ImdbMovie {
  @Column("character varying", { primary: true, name: "imdb_id", length: 100 })
  imdbId: string;

  @Column("real", { name: "rating", nullable: true, precision: 24 })
  rating: number | null;

  @Column("integer", { name: "vote", nullable: true })
  vote: number | null;

  @OneToMany(() => Movie, (movie) => movie.imdb)
  movies: Movie[];
}

File: engine/engine.api/Models/ImdbMovie.cs
using System;
using System.Collections.Generic;

namespace engine.api.Models
{
    public partial class ImdbMovie
    {
        public ImdbMovie()
        {
            Movies = new HashSet<Movie>();
        }

        public string ImdbId { get; set; } = null!;
        public float? Rating { get; set; }
        public int? Vote { get; set; }

        public virtual ICollection<Movie> Movies { get; set; }
    }
}
